src/components/number-value.js

```diff
@@ -1,10 +1,23 @@
 import AnyValue from './any-value'
 
 class NumberValue extends AnyValue {
-  constructor(...args) {
-    super(...args, 0)
-    this.define('increment', (v, n = 1) => v + n)
-    this.define('decrement', (v, n = 1) => v - n)
+  static defaultProps = {
+    max: Number.MAX_SAFE_INTEGER,
+    min: Number.MIN_SAFE_INTEGER,
+  }
+  constructor(props, ...rest) {
+    super(props, ...rest, 0)
+    const { max, min } = props
+
+    this.define('increment', (v, n = 1) => {
+      const num = typeof n !== 'number' ? 1 : n
+      return Math.min(v + num, max)
```

Comment:
Should this take `this.props.max` instead? To allow people to change the maximum/minimum over time.

Reply:
oh boy... yes absolutely.  good catch